experimental/dds/tree2/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -219,137 +213,13 @@ export const counter: BrandedFieldKind<
 	new Set(),
 );
 
-const valueRebaser: FieldChangeRebaser<ValueChangeset> = {
-	compose: (
-		changes: TaggedChange<ValueChangeset>[],
-		composeChildren: NodeChangeComposer,
-	): ValueChangeset => {
-		if (changes.length === 0) {
-			return {};
-		}
-		let newValue: NodeUpdate | undefined;
-		const childChanges: TaggedChange<NodeChangeset>[] = [];
-		for (const { change, revision } of changes) {
-			if (change.value !== undefined) {
-				newValue = change.value;
-
-				// The previous changes applied to a different value, so we discard them.
-				// TODO: Consider if we should represent muted changes
-				childChanges.length = 0;
-			}
-
-			if (change.changes !== undefined) {
-				childChanges.push(tagChange(change.changes, revision));
-			}
-		}
-
-		const composed: ValueChangeset = {};
-		if (newValue !== undefined) {
-			composed.value = newValue;
-		}
-
-		if (childChanges.length > 0) {
-			composed.changes = composeChildren(childChanges);
-		}
-
-		return composed;
-	},
-
-	amendCompose: () => fail("Not implemented"),
-
-	invert: (
-		{ revision, change }: TaggedChange<ValueChangeset>,
-		invertChild: NodeChangeInverter,
-		reviver: NodeReviver,
-	): ValueChangeset => {
-		const inverse: ValueChangeset = {};
-		if (change.changes !== undefined) {
-			inverse.changes = invertChild(change.changes, 0);
-		}
-		if (change.value !== undefined) {
-			assert(revision !== undefined, 0x591 /* Unable to revert to undefined revision */);
-			inverse.value = {
-				revert: reviver(revision, 0, 1)[0],
-				// KLUDGE: Value field changes don't have local IDs so we use an arbitrary value.
-				// TODO: Either delete the value field implementation or give its changes some local IDs.
-				changeId: { revision, localId: brand(42) },
-			};
-		}
-		return inverse;
-	},
-
-	amendInvert: () => fail("Not implemnted"),
-
-	rebase: (
-		change: ValueChangeset,
-		over: TaggedChange<ValueChangeset>,
-		rebaseChild: NodeChangeRebaser,
-	): ValueChangeset => {
-		if (change.changes === undefined || over.change.changes === undefined) {
-			return change;
-		}
-		return { ...change, changes: rebaseChild(change.changes, over.change.changes) };
-	},
-
-	amendRebase: (
-		change: ValueChangeset,
-		over: TaggedChange<ValueChangeset>,
-		rebaseChild: NodeChangeRebaser,
-	) => ({ ...change, changes: rebaseChild(change.changes, over.change.changes) }),
-};
-
 export interface ValueFieldEditor extends FieldEditor<ValueChangeset> {
 	/**
 	 * Creates a change which replaces the current value of the field with `newValue`.
 	 */
 	set(newValue: ITreeCursor): ValueChangeset;
 }
 
-const valueFieldEditor: ValueFieldEditor = {
```

Comment:
I think it's good that we're avoiding the code duplication between the value field editor and the optional field editor. It would be nice to the keep `ValueFieldEditor` interface just so that the ruse of using the optional field logic to implement the value field logic remain fully contained to this file. For example, right now the DefaultChangeFamily code is privy to the ruse and could mess it up by passing `true` instead of `false` to the `set` method. Even without thinking about maintainability, it's a little weird that the editor exposed on the value field takes this boolean that must always be `false` and accepts `undefined` content but it's invalid to pass it `undefined` .

I think you'd end up with something like this:
```
export interface ValueFieldEditor extends FieldEditor<OptionalChangeset> {
	/**
	 * Creates a change which replaces the current value of the field with `newValue`.
	 */
	set(newValue: ITreeCursor, id: ChangesetLocalId): OptionalChangeset;
}

const valueFieldEditor: ValueFieldEditor = {
	...optionalFieldEditor,
	set: (newContent: ITreeCursor, id: ChangesetLocalId): OptionalChangeset =>
		optionalFieldEditor.set(newContent, false, id),
};

const valueChangeHandler: FieldChangeHandler<OptionalChangeset, ValueFieldEditor> = {
	rebaser: optionalChangeRebaser,
	codecsFactory: makeOptionalFieldCodecFamily,
	editor: valueFieldEditor,

	intoDelta: (change: OptionalChangeset, deltaFromChild: ToDelta) =>
		optionalFieldIntoDelta(change, deltaFromChild),
	isEmpty: (change: OptionalChangeset) =>
		change.childChange === undefined && change.fieldChange === undefined,
};
```

Reply:
Nice, I agree that this is a lot cleaner. I did this as part of the latest commit.